### PR TITLE
Us 36 fix

### LINKED ISSUE #47: User Story 36, Merchant Dashboard Statistics
```
As a merchant
When I visit my dashboard, I see an area with statistics:
- top 5 items I have sold by quantity, and the quantity of each that I've sold
- total quantity of items I've sold, and as a percentage against my sold units plus remaining 
inventory (eg, if I have sold 1,000 things and still have 9,000 things in inventory, the message 
would say something like "Sold 1,000 items, which is 10% of your total inventory")
- top 3 states where my items were shipped, and their quantities
- top 3 city/state where my items were shipped, and their quantities (Springfield, MI should not be 
grouped with Springfield, CO)
- name of the user with the most orders from me (pick one if there's a tie), and number of orders
- name of the user who bought the most total items from me (pick one if there's a tie), and the 
total quantity
- top 3 users who have spent the most money on my items, and the total amount they've spent
```

DO THESE STORIES FIRST:
- User Story 34, Merchant Dashboard Show Page

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #233

child of #274

depends on #232

### PR DESCRIPTION
**Pull request by WHomer**
What does this PR do? 
- Resolves #47 
- fixes minor bugs/display issues.

### CODE REVIEW

File: app/views/merchants/orders/index.html.erb [hunk 1]
```diff
@@ -10,13 +10,96 @@
...
+<% end %>
+
 <% if current_admin?  %>
   <%= link_to "Downgrade to User", admin_merchant_edit_path({id: @user.id, request: :downgrade_to_user}), :method => :patch, class:"btn btn-primary" %>
 <% end %>
 
 <% @orders.each do |order| %>
 <section id="order-<%= order.id %>">
-  <p>Order id: <%= link_to "#{order.id}", dashboard_order_path(order.id) %>
+  <p>Order id: <%= link_to "#{order.id}", profile_order_path(order.id) %>
```
Comment (BrennanAyers) [inline app/views/merchants/orders/index.html.erb]: This needs to change back to dashboard_order_path, or else merchants will be unable to look at orders and fulfill items

File: app/models/user.rb [hunk 2]
```diff
@@ -10,6 +10,75 @@ class User < ApplicationRecord
...
 
+  def best_customer_items
+    order = OrderItem.joins("JOIN orders ON orders.id=order_items.order_id")
+             .joins(('JOIN users ON users.id=orders.user_id'))
+             .joins('JOIN items ON items.id=order_items.item_id')
+             .where("orders.status=2 AND items.user_id=#{self.id}")
+             .select('sum(order_items.quantity) AS total_bought, users.id as user_id, users.*')
+             .group('users.id')
+             .order('total_bought desc')
+             .limit(1)
```
Comment (BrennanAyers) [inline app/models/user.rb:21]: Y'all are gods.

File: app/views/merchants/orders/index.html.erb [hunk 3]
```diff
@@ -10,13 +10,96 @@
 <p>State: <%= @user.state %></p>
 <p>Zip Code: <%= @user.zip %></p>
 
+<% if !@user.items.empty? && !@user.top_items_sold(5).empty?  %>
```
Comment (BrennanAyers) [inline app/views/merchants/orders/index.html.erb:13]: 👍

**Review by BrennanAyers — APPROVED** [review]
Tests passing!
